fix/TagUrl to TS

dhruvv173 opened the pull request:
## Explanation

- Fixes #19126 

## Screenshots/Screencaps


### Before

https://github.com/MetaMask/metamask-extension/assets/79097544/324a8003-e369-4ebd-8bac-6b09e62a69e1

### After

https://github.com/MetaMask/metamask-extension/assets/79097544/bb7b2f18-829b-4a23-8ca3-223deec9d698

unit tests
![image](https://github.com/MetaMask/metamask-extension/assets/79097544/aba3b9ca-7778-438a-9c9c-adaa03559ab5)


## Manual Testing Steps

- Check storybook controls for `TagUrl` on this branch
- `yarn jest ui/components/component-library/tag-url`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

dhruvv173 committed:
TagUrl to TS

georgewrmarshall commented:
Hey @dhruvv173, could you please fix the type errors and failing snapshot tests and I will review. Let me know if you need any support. Thanks!

dhruvv173 committed:
updating snapshot and fixing type error

dhruvv173 commented on ui/components/component-library/tag-url/tag-url.tsx:
```diff
@@ -0,0 +1,90 @@
+        display={Display.Flex}
+        {...(props as BoxProps<C>)}
+      >
+        <AvatarFavicon src={src} name={label} {...avatarFaviconProps}>
+          {children}
+        </AvatarFavicon>
```
the TS error was due to not passing `children` here. 
@georgewrmarshall  Could you please check this when you do a full review and confirm if it is acceptable? Thank you!

georgewrmarshall replied:
Hey @dhruvv173, I think the move here is to omit it from `AvatarFavicon` as a prop. I've made that update everything else looks pretty good 💯

georgewrmarshall committed:
Ommiting the children prop from AvatarBaseStyleUtilityProps in AvatarFavicon so it is not a required prop
Also updating AvatarFavicon tests I think a rebase should resolve the issue with fallbackIconProps test

georgewrmarshall commented on ui/components/component-library/avatar-favicon/avatar-favicon.test.tsx:
```diff
@@ -99,7 +99,6 @@ describe('AvatarFavicon', () => {
     const { getByTestId } = render(
       <AvatarFavicon
         className="mm-avatar-favicon--test"
-        name="test"
```
Was getting a duplicate prop type error in the test file

georgewrmarshall commented on ui/components/component-library/avatar-favicon/avatar-favicon.types.ts:
```diff
@@ -12,7 +12,7 @@ export enum AvatarFaviconSize {
 
 export interface AvatarFaviconStyleUtilityProps
-  extends Omit<AvatarBaseStyleUtilityProps, 'size'> {
+  extends Omit<AvatarBaseStyleUtilityProps, 'size' | 'children'> {
```
Omitting children as we don't use allow `children` prop to be used in `AvatarFavicon`

georgewrmarshall commented on ui/components/component-library/tag-url/README.mdx:
```diff
@@ -120,9 +130,9 @@ import { TagUrl } from '../../ui/component-library/tag-url';
+  label="metamask.github.io"
 />
 <TagUrl
   src=""
```
Made some small updates to the docs page to show missing images and improve the sentence structure of some sections

georgewrmarshall commented on ui/components/component-library/tag-url/tag-url.types.ts:
```diff
@@ -0,0 +1,55 @@
+  /**
+   * It accepts all the props from ButtonLink
+   */
+  actionButtonProps?: ButtonLinkProps<'button'>;
```
Hmm I'm not sure how this component will act if we pass it an href or attributes related to an `a` tag. I've created some stories that use href and they seem to be working without issue so we can monitor it.

georgewrmarshall reviewed (approved):
LGTM! I've made some small updates during my review. Otherwise excellent work! Thanks for your contribution @dhruvv173 💯

garrettbear reviewed (approved):
Nice
